### DISCUSSION TITLE
ideas to make the libray more flexible

### DISCUSSION THREAD

**Discussion by softwarecrash** [maland16/daly-bms-uart#3] [Ideas]
hello,
a bunch of points to makle this great libray a little bit more flexibel
- add option to change form hardwareserial to softwareserial so it can used with esp82XX or other smal chips
- change printf to someting, not all arduino variants support this, or make a little function to support printf
- change checksum calculate away from the cheating method, so its possibil in future to send commands for controlling the fets
i will try to help you, but i strugle at hte point to implement software serial, its hard written in the class

**Comment by softwarecrash** [comment]
please close this, its done :)

**Comment by maland16** [comment]
Locking conversation because these features were implemented. Thanks @softwarecrash!!!
